### DISCUSSION TITLE
Milvus Standalone support S3 external with AWS

### DISCUSSION THREAD

**Discussion by eliyahu77** [milvus-io/milvus#28763] [Q&A and General discussion]
HI * , 

I want to deploy Milvus with externalS3 using service account with S3 Bucket on aws this is support by Standalone toplogy or I have to install Milvus with Cluster mode

**Comment by yhmo** [comment]
Milvus has a configuration file milvus.yaml. This file is under the path `/milvus/configs/milvus.yaml` of the the container of milvus server.
To modify this file, you need to map it to an outside file in your local, this doc tells how to do that: https://milvus.io/docs/configure-docker.md
After this file is mapped. You need to modify the outside file, change the configurations for S3, this doc tells how to do that: https://milvus.io/docs/deploy_s3.md#Configure-S3-with-Docker-Compose

After all the configurations are done, start the milvus standalone server.

**Reply by eliyahu77** [reply]
@yhmo  
I  Try to deploy it  with helm  on EKS there is any Docs that  point how to do it

**Reply by yhmo** [reply]
Configure the externalS3 section in the values.yaml file:
https://milvus.io/docs/deploy_s3.md#Configure-S3-on-K8s

**Reply by eliyahu77** [reply]
@yhmo  i did this config on eks 

with follwiing  helm command 

`helm install milvus milvus/milvus --namespace mynamespace --version 4.1.8 --set standalone.replicas=1,standalone.heaptrack.enabled=False,standalone.disk.enabled=True,standalone.disk.size.enabled=False,standalone.persistence.mountPath=/var/lib/milvus,standalone.persistence.enabled=True,standalone.persistence.persistentVolumeClaim.existingClaim=,standalone.persistence.persistentVolumeClaim.storageClass=mystorageclass,standalone.persistence.persistentVolumeClaim.accessModes=ReadWriteOnce,standalone.persistence.persistentVolumeClaim.size=50Gi,externalS3.enabled=True,externalS3.host=s3.region.amazonaws.com,externalS3.port=443,externalS3.useSSL=True,externalS3.bucketName=bucketnameonaws,externalS3.rootPath=passmilvus,externalS3.useIAM=True,externalS3.cloudProvider=aws,serviceAccount.create=false,serviceAccount.name=myserviceaccount,cluster.enabled=False,pulsar.enabled=False,etcd.replicaCount=1,etcd.persistence.storageClass=mystorageclass`

 i getting pod

 milvus-standalone-747f9bfb94-xpllx                 0/1     CrashLoopBackOff


# this log 
` [ERROR] [grpcclient/client.go:411] ["retry func failed"] ["retry time"=4] [error="empty grpc client: find no available querycoord, check querycoord state"] [stack="github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).call\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:411\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).Call\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:489\ngithub.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase[...]).ReCall\n\t/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:505\ngithub.com/milvus-io/milvus/internal/distributed/querycoord/client.wrapGrpcCall[...]\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/client/client.go:95\ngithub.com/milvus-io/milvus/internal/distributed/querycoord/client.(*Client).GetMetrics\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/querycoord/client/client.go:267\ngithub.com/milvus-io/milvus/internal/rootcoord.(*QuotaCenter).syncMetrics.func1\n\t/go/src/github.com/milvus-io/milvus/internal/rootcoord/quota_center.go:191\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\t/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75"]
`

#

`[error="Endpoint url cannot have fully qualified paths."] [stack="github.com/milvus-io/milvus/internal/datacoord.(*Server).newChunkManagerFactory\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/server.go:462\ngithub.com/milvus-io/milvus/internal/datacoord.(*Server).initDataCoord\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/server.go:335\ngithub.com/milvus-io/milvus/internal/datacoord.(*Server).Init\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/server.go:323\ngithub.com/milvus-io/milvus/internal/distributed/datacoord.(*Server).init\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/service.go:126\ngithub.com/milvus-io/milvus/internal/distributed/datacoord.(*Server).Run\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/service.go:249\ngithub.com/milvus-io/milvus/cmd/components.(*DataCoord).Run\n\t/go/src/github.com/milvus-io/milvus/cmd/components/da`

**Comment by yhmo** [comment]
[error="Endpoint url cannot have fully qualified paths."]   ---------------------- This error indicates the s3 url is improper. It is an error returned from s3 interfaces, milvus could not initialize the connection to s3. Search "Endpoint url cannot have fully qualified paths" on Google for more info.
